Deprecate workload

Harshil2107 opened the pull request.

Harshil2107 committed:
stdlib, resources: skeleton for workload resouce

Change-Id: I5017ac479ad61c767ede36fae195105e0519304f

Harshil2107 committed:
stdlib, resources: depricated workload

- Added WrokloadResource in resource.py.

- depricated Workload and CustomWorkload.

- changed iscvmatched-fs.py with obtain resource for workload to test.

Change-Id: I2267c44249b96ca37da3890bf630e0d15c7335ed
Note: change example files back to original

BobbyRBruce commented on src/python/gem5/resources/resource.py:
```diff
@@ -554,6 +554,81 @@ def _get_weights_and_simpoints_from_file(
+    def __init__(
+        self,
+        function: str = None,
+        resources: Optional[Dict[str, str]] = None,
+        additional_params: Optional[Dict[str, str]] = None,
```
Why we need these two lines? Isn't this what `parameters` is for?

Harshil2107 committed:
stdlib, resources: Added warn msgs and commets.

- Added deprecated warnings to Workload and Abstract workload.

- Added comments to the classes changed.

Change-Id: I671daacf5ef455ea65103bd96aa442486142a486

Harshil2107 committed:
stdlib, resources: Updated warn messages

- Updated warn messages to be more clear.

Change-Id: I2a8922e96b5b544f2f229b01b3c51fc5e79995b4

Harshil2107 committed:
stdlib, resources: Moved parsing params.

- moved parsing from WorkloadResource __init__ to obtain_resource

Change-Id: I9ed2aebb23af7f359bc1e5fff8ebe616a0da1374

Harshil2107 committed:
stdlib, tests, resources: Updated tests

- Updated workload tests to use WrokloadResource and obtain_resource

Change-Id: I39194e7fe764566a528e5141c29f30efa14e0cde

Harshil2107 committed:
stdlib: Changed use of Workload to obtain_resource

- Changed files calling Workload class to call obtain_resoucre instead.

Change-Id: I41f5f0c3ccc7c08b39e7049eabef9609d6d68788
